rest.go

```diff
@@ -90,39 +51,39 @@ func BuildResponse(res *http.Response) (*Response, error) {
 }
 
 // API is the main interface to the API.
-func API(request Request) (*Response, error) {
+func API(request *http.Request) (*Response, error) {
```

Comment:
That said, for the same reason that we're switching to `http.Request` for the input, it seems reasonable to switch to `*http.Response` for the returned value. rest.Response mainly just seems to be there to avoid the few lines involved in an `ioutil.ReadAll` call plus the error checking -- all the other fields exist in http.Response already (some of which are useful, and would require abandonment of this library to access)